inc/commondbtm.class.php

```diff
@@ -1161,6 +1161,87 @@ function add(array $input, $options = [], $history = true) {
       return false;
    }
 
+   /**
+    * Clones the current item
+    *
+    * @param boolean $history do history log ? (true by default)
+    *
+    * @return integer the new ID of the clone (or false if fail)
+    */
+   function clone($history = true) {
+      global $DB, $CFG_GLPI;
+
+      if ($DB->isSlave()) {
+         return false;
+      }
+
```

Comment:
I wonder if the following code coul be sufficient:

```
$new_item = new static();
$input = Toolbox::addslashes_deep($this->fields);
$input['_oldID'] =  $this->fields['id'];
unset($input['id']);
return $new_item->add($input);

Reply:
I don't completely understand what $input contains. Will this work on sub-elements on computers ? (like devices, OS, notes, software, ....)

Reply:
`$input` contains form entries (on creation and on edition) which should have been changed by previous methods. `CommonDBTM` is designed to handle all objects in GLPI that have a table in the database, so yes; that should work for almost everything.

Cedric's proposal would indeed be better, since it relies on existing and well tested code instead of duplicate it (well, sort of).
It just miss the dates reset as far as I can see.

Reply:
Ok I see, I will update the code and add the date reset to it